realm/realm-library/src/androidTestObjectServer/java/io/realm/SessionTests.java

```diff
@@ -1,583 +0,0 @@
-        }
-    }
-
-
-    @Test
-    public void downloadAllServerChanges_withTimeout_invalidParametersThrows() throws InterruptedException {
-        Realm realm = Realm.getInstance(configuration);
-        SyncSession session = app.getSync().getSession(configuration);
-        try {
-            try {
-                session.downloadAllServerChanges(-1, TimeUnit.SECONDS);
-                fail();
-            } catch (IllegalArgumentException ignored) {
-            }
-
-            try {
-                //noinspection ConstantConditions
-                session.downloadAllServerChanges(1, null);
-                fail();
-            } catch (IllegalArgumentException ignored) {
-            }
-        } finally {
-            realm.close();
-        }
-    }
-
-    @Test
-    public void downloadAllServerChanges_returnFalseWhenTimedOut() throws InterruptedException {
-        Realm realm = Realm.getInstance(configuration);
-        SyncSession session = app.getSync().getSession(configuration);
-        try {
-            assertFalse(session.downloadAllServerChanges(100, TimeUnit.MILLISECONDS));
-        } finally {
-            realm.close();
-        }
-    }
-
-    @Test
-    @UiThreadTest
-    public void unrecognizedErrorCode_errorHandler() {
-        AtomicBoolean errorHandlerCalled = new AtomicBoolean(false);
-        configuration = configFactory.createSyncConfigurationBuilder(user)
-                .errorHandler((session, error) -> {
-                    errorHandlerCalled.set(true);
-                    assertEquals(ErrorCode.UNKNOWN, error.getErrorCode());
-                    assertEquals(ErrorCode.Category.FATAL, error.getCategory());
-
-                })
-                .build();
-        Realm realm = Realm.getInstance(configuration);
-        SyncSession session = app.getSync().getSession(configuration);
-
-        TestHelper.TestLogger testLogger = new TestHelper.TestLogger();
-        RealmLog.add(testLogger);
-
-        session.notifySessionError("unknown", 3, "Unknown Error");
-        RealmLog.remove(testLogger);
-
-        assertTrue(errorHandlerCalled.get());
-        assertEquals("Unknown error code: 'unknown:3'", testLogger.message);
-
-        realm.close();
-    }
-
-    @Test
-    public void getSessionThrowsOnNonExistingSession() {
-        Realm realm = Realm.getInstance(configuration);
-        SyncSession session = app.getSync().getSession(configuration);
-        assertEquals(configuration, session.getConfiguration());
-
-        // Closing the Realm should remove the session
-        realm.close();
-        try {
-            app.getSync().getSession(configuration);
-            fail("getSession should throw an ISE");
-        } catch (IllegalStateException expected) {
-            assertThat(expected.getMessage(), CoreMatchers.containsString(
-                    "No SyncSession found using the path : "));
-        }
-    }
-
-    @Test
-    public void isConnected_falseForInvalidUser() {
```

Comment:
@cmelchior What should this test do? What is an invalid user, _logged out_ maybe? 

Further, in the current form it is nondeterministic, due to not awaiting connection state, etc. 

Could you sketch suggested test if it still makes sense

Reply:
Removed the test as we per discussions cannot create an invalid user here.